lib/modules/platform/gerrit/utils.ts

```diff
@@ -0,0 +1,111 @@
+import { CONFIG_GIT_URL_UNAVAILABLE } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import type { BranchStatus, PrState } from '../../../types';
+import * as hostRules from '../../../util/host-rules';
+import { parseUrl } from '../../../util/url';
+import { hashBody } from '../pr-body';
+import type { Pr } from '../types';
+import type {
+  GerritChange,
+  GerritChangeStatus,
+  GerritLabelTypeInfo,
+} from './types';
+
+export const TAG_PULL_REQUEST_BODY = 'pull-request';
+
+export function getGerritRepoUrl(repository: string, endpoint: string): string {
+  // Find options for current host and determine Git endpoint
+  const opts = hostRules.find({
+    hostType: 'gerrit',
+    url: endpoint,
+  });
+
+  const url = parseUrl(endpoint);
+  if (!url) {
+    throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+  }
+  url.username = opts.username ?? '';
+  url.password = opts.password ?? '';
```

Comment:
isn't it an error If one of them are missing?

Reply:
correct, i can't imagine a working scenario without a username. 
And additionally it will always be fail previously from the `initPlatform` call.